src/main/java/org/grouplens/grapht/reflect/internal/ClassInstantiator.java

```diff
@@ -98,38 +107,59 @@ public Object instantiate() throws ConstructionException {
         } catch (IllegalAccessException e) {
             throw new ConstructionException(ctor, "Access violation on " + ctor, e);
         }
+        finally {
+            mdcContext.finish();
```

Comment:
Now that I think about things more clearly, the calls to the setter and field injection points below should also be in the same MDC context. So it should probably have its own `try`/`finally` that wraps most of the method. This might be the time to split this method into helper methods.